### PR TITLE
AK+Spreadsheet: Fix number to column index conversion functions

### PR DESCRIPTION
**Pull request by tcl3**
This PR corrects the output of `DeprecatedString::bijective_base_from()` and `Spreadsheet::convert_from_string()` for numbers larger than `base^2`.

This fixes the display of column index values in Spreadsheet.

spreadsheet_column_index_broken

NB: AFAICT the test case for `bijective_base_from()` value "ABC" was incorrect. I have updated it, as well as adding more test cases.

### CODE REVIEW

File: Userland/Applications/Spreadsheet/Spreadsheet.cpp [hunk 1]
```diff
@@ -98,20 +98,20 @@ static Optional<size_t> convert_from_string(StringView str, unsigned base = 26,
...
     for (auto i = 0u; i < len; i++) {
         auto maybe_index = map.find(str[i]);
         if (!maybe_index.has_value())
             return {};
         size_t digit_value = maybe_index.value();
-        // NOTE: Refer to the note in `String::bijective_base_from()'.
-        if (i == 0 && len > 1)
-            ++digit_value;
-        value += digit_value * AK::pow<float>(base, len - 1 - i);
+        value += (digit_value + 1) * AK::pow<float>(base, len - 1 - i);
```
Comment (konradekk) [inline Userland/Applications/Spreadsheet/Spreadsheet.cpp:111]: itʼd be nice to have a test for that too…

as far as i understood it should be enough to assert that composing `convert_from_string` with `bijective_base_from` on both sides (for few relevant cases) is an identity map… (as we probably care more for those to be inverses than anything else)  🤔

**Review by alimpfard — COMMENTED** [review]
Fixes look right, but there's also an implementation of this in https://github.com/SerenityOS/serenity/blob/f8bdb584f8552f553c106fc7b32d5129913d8cd0/Userland/Applications/Spreadsheet/Tests/mock.test-common.js#L18 used for Spreadsheet tests, it'd be nice if you could fix that one too (if it needs fixing) :)

**Comment by tcl3** [comment]
@alimpfard - I've tested the `toBijectiveBase()` function with the same test inputs used in the `bijective_base_from` unit tests. It already gives the same output as my fixed version, so I don't believe anything needs doing.

**Comment by konradekk** [comment]
(to be frank: i donʼt believe this encoding belongs to the string class – iʼve seen usage of that only in the spreadsheet – but this could only mean i havenʼt seen much! 😅)

**Comment by alimpfard** [comment]
@konradekk it's also used in LibWeb to produce names for ordered lists.

**Pull request merged**